[CS2113T-T10-2] TermiNUS

kelvneo opened the pull request:
TermiNUS is a CLI (command line interface) program for NUS Students who wish to organize their NUS academic materials through a CLI. The product aims to aid student in organizing their academic schedule and enhancing their learning experiences.

3m0W33D committed:
Merge pull request #131 from LouisLouis19/branch-BugFixes

Bug Fixes

Woolicious98 committed:
Merge pull request #128 from Woolicious98/branch-Note-FileStorage

Bug fix for file permissions

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-test

kelvneo committed:
Add Ui Class Diagram and Explanation.

Update Active Recall paragraph as well.

Commit:
Add check lower case for note name

3m0W33D committed:
Merge branch 'master' into upstream/master

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

kelvneo committed:
Merge pull request #133 from kelvneo/master

Add Ui Class Diagram and Explanation

Commit:
Add JavaDocs for isDuplicateName

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

Woolicious98 committed:
Merge pull request #134 from Woolicious98/branch-test

Fix note check for lower case

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-DG

Commit:
Update DG with storage and add trim to file path checks

Woolicious98 committed:
Merge pull request #112 from Woolicious98/branch-DG

Add storage implementation in DG

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

3m0W33D committed:
Update DG table and fix class diagrams

3m0W33D committed:
Update typo

3m0W33D committed:
Merge pull request #136 from 3m0W33D/master

Update broken table and fix class diagram for DG

pos0414 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+Initialization of all the classes required for Terminus class to run
+
+![](attachments/MainLogic.png)
```
Is this format correct? Should there be a `:` in front of the class name?
![image](https://user-images.githubusercontent.com/47795460/138995874-cd758cdc-d586-4ad9-a6e0-92b24e5e5daf.png)

pos0414 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+- accessing the arraylist of contents
+
+### 3.7 Active Recall Component
+![Active Recall Class Diagram](attachments/ActiveRecallClassDiagram.png)
```
Since there is already a direct association in the same direction as the dependence, can the dashed line be omitted? 
![image](https://user-images.githubusercontent.com/47795460/138996519-3af93c3b-e92f-4cc6-a2a0-c044aaf18905.png)

pos0414 reviewed (commented):
Just some minor issues with sequence diagrams of your DG. Other than that, looks good!

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

SuibianP commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+#### 2.1.2 Getting the project files
+
+Go to [link](https://github.com/AY2122S1-CS2113T-T10-2/tp) and retrieve the `TermiNUS project file`.
+You can do so by **forking** the project and **cloning** a copy into your computer.
```
Would it be better to just mention "clone", since it is not compulsory that others fork the project to peek into it?

SuibianP commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+> 💡 IntelliJ IDEA have certain shortcut key to aid in auto-formatting of code.
+> Once you are done with a piece of code, highlight the section you have just written and press the
+> key `CTRL + SHIFT + L`.
```
For keys, would you consider using `<kbd>` HTML tags so that they stand out better? An example: <kbd>Ctrl + Shift + L</kbd>

SuibianP commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+
+### 3.3 Parser Component
+![](attachments/ParserClassDiagram.png)
```
Would it be better to disable the icons, as per [Prof's words](https://github.com/nus-cs2113-AY2122S1/forum/issues/114)?

SuibianP commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+
+### 3.4 Command Component
+![](attachments/CommandClassDiagram.png)
```
For UML diagrams, it is also a solution to use online endpoints to generate the image from your source on the fly. This reduces the need to have binary files and keeps everything in one place.

ngnigel99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+### 3.1 Architecture
+
+![](attachments/MainInit.png)  
```
Is this segment missing elaboration? I think it would be better to give a brief summary or explanation of the diagram

ngnigel99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+Initialization of all the classes required for Terminus class to run
+
+![](attachments/MainLogic.png)
```
Is the `ref` portion of this diagram necessary? I'm not sure how it interacts with the whole system
![image](https://user-images.githubusercontent.com/69423287/139171876-1728f87c-3a24-4b9d-90cd-3054c23cebb5.png)

ngnigel99 replied:
Can the note part here be replaced with something more representative of a sequence diagram?
![image](https://user-images.githubusercontent.com/69423287/139171985-88f96ed1-b2f4-4e3d-ac50-8b6d11a8af14.png)

ngnigel99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+### 3.5 Module Component
+
+![](attachments/Module.png)
```
I think this could just be a `*`
![image](https://user-images.githubusercontent.com/69423287/139172206-bcd43957-53a9-4247-b082-3cb6ac49b659.png)

ngnigel99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+### 3.8 Storage Component
+
+
+![](attachments/StorageComponent.png)
```
Similar to the previous recommendation, I think this could just be a `*`
![image](https://user-images.githubusercontent.com/69423287/139172403-afb2bbc6-e0dd-4432-8e08-9ab1665a5b0b.png)

ngnigel99 reviewed (commented):
(DG review during tutorial)
Overall, great effort in producing the diagrams and elaboration. However, I think there are some minor changes you guys can make to improve the diagrams!

KZQ1999 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+#### 4.2.1 Current Implementation
+
+![Active Recall Sequence Diagram](attachments/ActiveRecallSequenceDiagram.png)
```
Could the sequence diagram be broken down into different parts using reference frame, as it might be too big to follow

KZQ1999 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+- Provides functionality to list all commands for the help `Command`
+
+
+### 3.4 Command Component
```
Perhaps the notation for visibility should be + and - as taught instead of green circles and boxes

leeyikai commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+## 3. Design
+
+### 3.1 Architecture
```
Would be better to have a brief explanation of what is going on! A brief intro of the different parts will be great.

leeyikai commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+### 3.2 UI Component
+![](attachments/UIClassDiagram.png)
+
+The Ui Component consists of the `Ui` class which handles all input and output operations within 
+TermiNUS application. To reduce coupling, we have used `Ui` on only the main runner `Terminus`, and 
+the Active Recall `GameEnvironment`. If future features require the extended use of `Ui`, they may 
+call `getInstance()` from `Ui` to get the same singleton class as both `GameEnvironment` and 
+`Terminus`.
```
Clear and good explanation! Very easy to follow

KZQ1999 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+To view the high-level diagram, head to [3.8 Storage](#38-storage-component).
+
+#### 4.5.1 Initialize Storage Implementation
```
good use of reference frame to simplify the diagram !

leeyikai commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+will be created to generate the `Question` order.  
+The `DifficultyModifier` is a utility class used to calculate and tweak the weights of `Question`
+after the user has provided feedback on the difficulty of the question. It uses a 
+[logistic curve](https://en.wikipedia.org/wiki/Logistic_function) to calculate the change in weight.
```
Might be good to have bullet points at the bottom to guide the reader through as this is quite a lot of info

KZQ1999 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+- Printing string arrays to the output through `printSection()`.
+
+
+### 3.3 Parser Component
```
Perhaps you should not use the circular icon beside the class instances and use : instead

KZQ1999 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+#### 4.1.1 Current Implementation
+The following sequence diagram shows how the timetable feature works:
+![](attachments/TimetableSequenceDiagram.png)
```
I believe theres something wrong with the nusmodule activation bar returning a module without being called first
the leftward dashed arrow.

leeyikai commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,561 @@
+
+![](attachments/desmos-graph.png)
+The parameters of the logistic curve can be viewed here:
+[https://www.desmos.com/calculator/qefovvnuhx](https://www.desmos.com/calculator/qefovvnuhx).
```
Clear and great explanation!

KZQ1999 reviewed (commented):
Overall pretty good job with the diagrams , just make it smaller so its easier for users and try to follow the UML notations in lectures

leeyikai reviewed (commented):
Looks great overall, just minor changes

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Update add, delete and reload note command

Commit:
Update delete and update module command

Commit:
Update Terminus to use StorageManager and add the rest of the storage types

Commit:
Update pdf storage

Commit:
Update load function in StorageManager

Commit:
Fix display messages

Commit:
Add filter manager

3m0W33D committed:
Merge branch 'master' into upstream/master

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Refactor Code

* Remove magic literal
* Remove repeat code

3m0W33D committed:
Refactor magic literal 

Fix #141

3m0W33D committed:
Add JUnit test for codecov

3m0W33D committed:
Fix JUnit test

Commit:
Add filter to load notes

Commit:
Update export command

Commit:
Fix checkstyle

3m0W33D committed:
Tighten regex for argument parsing and fix junit

3m0W33D committed:
Update text-ui-testing

Commit:
Add javadocs to newly created storage classes

3m0W33D committed:
Update AddNoteCommand Error message

3m0W33D committed:
Update JUnit test

3m0W33D committed:
Merge pull request #144 from 3m0W33D/branch-2.1-refactor

Command Error messages fix and Refactoring

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-File-Refactor

# Conflicts:
#	src/main/java/terminus/command/module/AddModuleCommand.java
#	src/main/java/terminus/command/module/DeleteModuleCommand.java
#	src/main/java/terminus/command/module/UpdateModuleCommand.java
#	src/main/java/terminus/common/Messages.java

Commit:
Remove modulestorage from Junit

Commit:
Isolate CommandResult extraction into StorageManager

Commit:
Add JUnit for createfolder and createfile

Commit:
Add checks to test if folder exists in JUnit

Commit:
Add after each to remove folder

Commit:
Add JUnit for Storage class

Commit:
Test enable on OS JUnit feature

Commit:
Add os dependencies tag for JUnit storage

Commit:
Add JUnit for NoteStorage

Commit:
Add JUnit for FolderStorage

Commit:
Add JUnit for PdfStorage

Commit:
Add JUnit to JsonStorage

Commit:
Add JUnit for StorageManager

Commit:
Add JUnit for FilterManager

Commit:
Remove ModuleStorage

Commit:
Fix error in JUnit for windows

Commit:
Fix windows JUnitr crash

Commit:
Add Terminus Logger to Storage

Commit:
Update filter for schedule

Commit:
Fix indentation erorrand optimise imports

Commit:
Add hadDUrationOverflow check for Filter

Commit:
Extract note filter to its own function

Commit:
Updat filter manager method to public static for checks

Woolicious98 committed:
Merge pull request #143 from Woolicious98/branch-File-Refactor

SLAP file storage

3m0W33D committed:
Update and fix PE Bugs

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Merge from master

kelvneo committed:
Update Developer Guide

- Include Missing Headings
- Update Active Recall Implementation
- Add Acknowledgements

kelvneo committed:
Fix PE-D issues and refactor Terminus

kelvneo committed:
Add TerminusTest

kelvneo committed:
Update JUnit tests

kelvneo committed:
Create codecov.yml

LouisLouis19 committed:
Ignore leading and trailing whitespace in add link arguments

LouisLouis19 committed:
SLAP parseArguments methods in AddLinkCommand class

LouisLouis19 committed:
Add duration in UG. Fix #169

LouisLouis19 committed:
Change the duration parameter description in UG. Fix #162

3m0W33D committed:
Change IText 7

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Merge branch 'master' into branch-2.1-refactor

LouisLouis19 committed:
Inform user on what are the expected arguments to be entered in error messages in UG. Fix #175

LouisLouis19 committed:
Add optional index example for view in UG. Add valid zoom_link instruction for add link command in UG.

3m0W33D committed:
Update text-ui-testing

kelvneo committed:
Merge pull request #184 from kelvneo/dg

Update Developer Guide

3m0W33D committed:
Attempt fix linux JUnit test

3m0W33D committed:
Fix UI-test

3m0W33D committed:
UI-text space fix

3m0W33D committed:
Update codecov

kelvneo committed:
Merge pull request #188 from LouisLouis19/branch-BugFixes

Fix PED Bugs and Address TA Code Review

kelvneo committed:
Merge pull request #186 from kelvneo/refactor

Fix PE-D issues and refactor Terminus

kelvneo committed:
Update codecov version

kelvneo committed:
Merge pull request #187 from 3m0W33D/branch-2.1-refactor

Fix PE Bugs and Update IText 5 to IText7

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

kelvneo committed:
Reorganize User Guide

kelvneo committed:
Add additional tip for commands

kelvneo committed:
Add extra information on tips and Q&A

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-DG

Commit:
Update Storage class diagram for DG

kelvneo committed:
Update UG based on feedback on PR

Commit:
Update Storage component in DG with FilterManager

Commit:
Add initialise storage in DG

Commit:
Add loading sequence diagram for storage in DG

3m0W33D committed:
Update DG with Architecture Diagram and sequence diagrams

3m0W33D committed:
Update with User Stories

3m0W33D committed:
Update with Non-Functional Requirements

Commit:
Add executeCommandResult seq diagram in DG

3m0W33D committed:
Create style.puml and update

Commit:
Add adding content and deleting content to DG

LouisLouis19 committed:
Add appendix E: Instructions for Manual Testing

LouisLouis19 committed:
Add timetable valid command input

kelvneo committed:
Merge pull request #189 from kelvneo/ug

Reorganize User Guide

3m0W33D committed:
Update Workspace implement and include puml style

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Merge branch 'master' into branch-DG-2.1-Update

3m0W33D committed:
Update NFD

3m0W33D committed:
Merge pull request #191 from 3m0W33D/branch-DG-2.1-Update

Update DG include Architecture diagram

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge branch 'master' into branch-DG

Commit:
Update JUnit for NoteStorage

Commit:
Update old stuff in DG

3m0W33D committed:
Update int overflow error message + include library

LouisLouis19 committed:
Fix sequence diagram, add design considerations

LouisLouis19 committed:
Fix add schedule format shown on error message

Commit:
Update puml to use the stylesheet

Commit:
Update DG with additional info

Commit:
Add value proposition

LouisLouis19 committed:
Draft PPP

LouisLouis19 committed:
Merge pull request #192 from LouisLouis19/Branch-v2.1

Update DG and draft PPP

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-DG

Commit:
Update timetable diagram with colors

Commit:
Update links in DG

Woolicious98 committed:
Merge pull request #190 from Woolicious98/branch-DG

Update DG Storage Components and Implementation

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Merge branch 'master' into branch-UpdateErrorMsg

3m0W33D committed:
Update typo in CommandExecution

3m0W33D committed:
Update ui text testing

3m0W33D committed:
Update DG dependencies

3m0W33D committed:
Add PPP

LouisLouis19 committed:
Fix Sequence Diagrams

kelvneo committed:
Add Project Portfolio page

LouisLouis19 committed:
Fix PPP draft format

kelvneo committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp into ppp

3m0W33D committed:
Update PPP Draft

kelvneo committed:
Update links of AboutUs

Commit:
Add PPP

kelvneo committed:
Change Overview of PPP

3m0W33D committed:
Update PPP

LouisLouis19 committed:
Fix links in PPP

kelvneo committed:
Merge pull request #194 from kelvneo/ppp

Add Project Portfolio page

Woolicious98 committed:
Merge pull request #196 from Woolicious98/juenwoo-PPP

Add my ppp

3m0W33D committed:
Finish PPP

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Merge branch 'master' into branch-UpdateErrorMsg

kelvneo committed:
Merge pull request #195 from LouisLouis19/Branch-v2.1

Fix Sequence Diagram and PPP format

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Merge branch 'master' into branch-UpdateErrorMsg

3m0W33D committed:
Change description and fix update module message

3m0W33D committed:
Merge pull request #193 from 3m0W33D/branch-UpdateErrorMsg

Update error msg and fix to include depdencies for IText 7

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Update overview

3m0W33D committed:
Merge branch 'master' into branch-UpdateErrorMsg

3m0W33D committed:
Merge pull request #197 from 3m0W33D/branch-UpdateErrorMsg

Update Overview for PPP

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Update to use OpenPDF and remove IText 7 and dependencies

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Fix bug in filtermanager

Woolicious98 committed:
Merge pull request #198 from Woolicious98/branch-Demo

Fix bug in filtermanager

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Update filtermanager with more cases

3m0W33D committed:
Merge branch 'master' into branch-OpenPDF

Commit:
Update incorrect json file

Commit:
Remove redundant code in storage

3m0W33D committed:
Update DG and PPP

3m0W33D committed:
Merge pull request #200 from 3m0W33D/branch-OpenPDF

Change PDF export to OpenPDF

Woolicious98 committed:
Merge pull request #199 from Woolicious98/branch-Demo

Update filtermanager with more cases

Commit:
Add set disabled in Terminus

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-Demo

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Make lowercase PPP file name

3m0W33D committed:
Merge pull request #202 from 3m0W33D/master

Change PPP file name

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-UG

Commit:
Update UG with name restirction

Woolicious98 committed:
Merge pull request #203 from Woolicious98/branch-UG

Update UG to provide some name restriction

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-fixes

Commit:
Fix minor bugs in UG

Commit:
Update and synchronise format for commands in application and UG

Commit:
Update text ui

Commit:
Add limitation of puml in DG

Commit:
Add storage disabled warnings

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Add extra note in DG

LouisLouis19 committed:
Update Sequence Diagram

kelvneo committed:
Update Sequence Diagrams and fix JUnit test

Woolicious98 committed:
Merge pull request #204 from Woolicious98/branch-fixes

Fix minor bugs in UG and DG

Woolicious98 committed:
Merge pull request #205 from 3m0W33D/branch-DG-fix

Minor update to DG

Woolicious98 committed:
Merge pull request #206 from LouisLouis19/Branch-v2.1

Update Sequence Diagram

3m0W33D committed:
Minor update to Command description

Woolicious98 committed:
Merge pull request #201 from Woolicious98/branch-Demo

Final setDisabled

kelvneo committed:
Merge pull request #207 from kelvneo/master

Update Sequence Diagrams and fix JUnit test

3m0W33D committed:
More description updates

3m0W33D committed:
Merge pull request #208 from 3m0W33D/branch-DG-fix

Minor update to Command description

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-test

Commit:
Update UG icons and language

kelvneo committed:
Fix diagrams for DG

kelvneo committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

kelvneo committed:
Update ConflictManager DG

Woolicious98 committed:
Merge pull request #209 from Woolicious98/branch-test

Update UG final

Woolicious98 committed:
Merge pull request #210 from kelvneo/master

Fix diagrams for DG

Commit:
Fix TOC in DG

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-test

Woolicious98 committed:
Merge pull request #211 from Woolicious98/branch-test

Fix TOC in DG

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Merge branch 'master' into branch-test

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Add copy in UG

Woolicious98 committed:
Merge pull request #212 from Woolicious98/branch-test

Update UG

kelvneo committed:
Change Logger error behaviour and separate StorageManager from Logger

kelvneo committed:
Merge pull request #213 from kelvneo/logger-fix

Change Logger error behaviour and separate StorageManager from Logger

3m0W33D committed:
Merge remote-tracking branch 'upstream/master'

3m0W33D committed:
Update UG and DG

3m0W33D committed:
Remove Page break

3m0W33D committed:
Update DG

3m0W33D committed:
Update Conflict

3m0W33D committed:
Add page break

3m0W33D committed:
Update page breaks

3m0W33D committed:
Remove extra lines

3m0W33D committed:
Update page break for UG

3m0W33D committed:
Update UG page break

3m0W33D committed:
Merge pull request #214 from 3m0W33D/master

Update DG

LouisLouis19 committed:
Update naming in sequence diagram

LouisLouis19 committed:
Merge pull request #215 from LouisLouis19/Branch-v2.1

Update naming in sequence diagram

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T10-2/tp

Commit:
Update UG with rationale

Woolicious98 committed:
Merge pull request #216 from Woolicious98/master

Update UG with rationale for CS2101
